### PR TITLE
Implement Smithing Templates

### PR DESCRIPTION
**Pull request by pandaaaBE**
## Introduction
This PR will add all Smithing Templates for a comming Armor Trim PR

(Is already finished)

https://imgur.com/hWklyfq.png

### API changes
```
VanillaItems::NETHERITE_UPGRADE_SMITHING_TEMPLATE()
VanillaItems::COAST_ARMOR_TRIM_SMITHING_TEMPLATE()
VanillaItems::DUNE_ARMOR_TRIM_SMITHING_TEMPLATE()
VanillaItems::EYE_ARMOR_TRIM_SMITHING_TEMPLATE()
VanillaItems::HOST_ARMOR_TRIM_SMITHING_TEMPLATE()
VanillaItems::RAISER_ARMOR_TRIM_SMITHING_TEMPLATE()
VanillaItems::RIB_ARMOR_TRIM_SMITHING_TEMPLATE()
VanillaItems::SENTRY_ARMOR_TRIM_SMITHING_TEMPLATE()
VanillaItems::SNOUT_ARMOR_TRIM_SMITHING_TEMPLATE()
VanillaItems::SPIRE_ARMOR_TRIM_SMITHING_TEMPLATE()
VanillaItems::TIDE_ARMOR_TRIM_SMITHING_TEMPLATE()
VanillaItems::VEX_ARMOR_TRIM_SMITHING_TEMPLATE()
VanillaItems::WARD_ARMOR_TRIM_SMITHING_TEMPLATE()
VanillaItems::WAYFINDER_ARMOR_TRIM_SMITHING_TEMPLATE()
VanillaItems::WILD_ARMOR_TRIM_SMITHING_TEMPLATE()
```

### CODE REVIEW

**Comment by pandaaaBE** [comment]
Why is "CI / Generated Code consistency checks" failed?

**Comment by jasonw4331** [comment]
It failed because you didn't autogenerate the phpdocs like the instructions say to

**Review by jasonw4331 — CHANGES_REQUESTED** [review]
VanillaItems doc comments must be autogenerated using the build scripts instead of being done by hand.

**Comment by pandaaaBE** [comment]
> VanillaItems doc comments must be autogenerated using the build scripts instead of being done by hand.

image...

**Comment by pandaaaBE** [comment]
Nvm I think i uploaded the wrong file

File: src/item/VanillaItems.php [hunk 1]
```diff
@@ -644,4 +662,24 @@ private static function registerArmorItems() : void{
 		self::register("netherite_leggings", new Armor(new IID(Ids::NETHERITE_LEGGINGS), "Netherite Leggings", new ArmorTypeInfo(6, 556, ArmorInventory::SLOT_LEGS, 3, true, material: ArmorMaterials::NETHERITE()), [EnchantmentTags::LEGGINGS]));
 	}
 
+	private static function registerSmithingTemplates() : void{
+		self::register("netherite_upgrade_smithing_template", new Item(new ItemIdentifier(ItemTypeIds::NETHERITE_UPGRADE_SMITHING_TEMPLATE), "Smithing Template"));
```
Comment (dktapps) [inline src/item/VanillaItems.php]: `
```suggestion
		self::register("netherite_upgrade_smithing_template", new Item(new IID(Ids::NETHERITE_UPGRADE_SMITHING_TEMPLATE), "Smithing Template"));
```
The imports are aliased for readability

Comment (dktapps) [reply src/item/VanillaItems.php]: Also, why is this just called "smithing template"? Shouldn't it have a more specific name?

**Pull request merged**